src/soroban.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Soroban helper class
+ * formatting, parsing, and etc
+ * @class Soroban
+ */
+export class Soroban {
+  /**
+   * All arithmetic inside the contract is performed on integers to
+   * avoid potential precision and consistency issues of floating-point
+   *
+   * This function takes the smart contract value and its decimals (if the token has any) and returns a display value
+   * @param {bigint | number | string} amount - the token amount you want to display
+   * @param {number} decimals - specify how many decimal places a token has
+   * @returns {string} - display value
+   */
+  static formatTokenAmount(amount, decimals) {
+    let formatted = amount.toString();
+
+    if (decimals > 0) {
+      formatted = (amount / 10 ** decimals).toFixed(decimals).toString();
+
+      // Trim trailing zeros
+      while (formatted[formatted.length - 1] === '0') {
+        formatted = formatted.replace(/0+$/, '');
+      }
+
+      if (formatted.endsWith('.')) {
+        formatted = formatted.replace(/0+$/, '');
+      }
+    }
+
+    return formatted;
+  }
+
+  /**
+   * parse token amount to use it on smart contract
+   *
+   * * This function takes the display value and its decimals (if the token has any) and returns an integer that'll be used it on smart contract
+   * @param {string} value - the token amount you want to use it on smart contract
+   * @param {number} decimals - specify how many decimal places a token has
+   * @returns {string} - smart contract value
+   *
+   *
+   * @example
+   * const displayValueAmount = "123.4560"
+   * const parsedAmountForSmartContract = StellarBase.Soroban.parseTokenAmount(displayValueAmount, 5); // "12345600"
```

Comment:
is this example okay? @Shaptic

Reply:
yeah, great! minor suggestion for readability

```suggestion
   * const parsedAmountForSmartContract = parseTokenAmount("123.4560", 5);
   * parsedAmountForSmartContract === "12345600"
```